### PR TITLE
Refactor/adding comments

### LINKED ISSUE #255: Add comments to frontend files
**Describe the requirement**
Make sure all frontend files have a satisfactory amount of comments so future contributors are able to understand how it works.

**Target audience**
 - Developers
 - Testers

### Issue Approval:
- [ ]  Issue approved by:

### PR DESCRIPTION
**Pull request by jge385**
# Description
This PR fixes #255

Add comments to the frontend code

# Type of change:
* Breaking change (This change would cause existing functionality to break)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

**Review by Jmcl168 — APPROVED** [review]
Looks good! These comments will definitely be helpful.

File: frontend/src/Components/Player.jsx [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
 
+    // check if players themselves are dead
+    // if players themselves are dead, they cannot hover any player circles
     const amIDead = !gameState.alivePlayers.includes(generalState.nickname);
 
+    // the hoverable variable depends on the following conditions:
+    // the game state [role or discussion or trial or undefined]
+    // the players are votable
+    // the players themselves are not dead
+    // the hovered players are not dead
```
Comment (Dhors) [inline frontend/src/Components/Player.jsx:33]: Nice commenting on the domain models

File: frontend/src/Components/Table.jsx [hunk 2]
```diff
@@ -33,6 +33,15 @@ export default function Table() {
...
 
+    /*
+        Because the players need to be sitting at arbitary points around the table,
+        absolute positioning is used relative to the table div. Each player is arranged
+        around the table such that they are at an equal angle apart from each other. 
+        The positions of each player (x, y) is calculated using the equations of 
+        an ellipse (as the table is an ellipse). The playerCoordinates are tracked in 
+        the playerCoords state variable. This variable changes is updated when the 
+        window is resized.
+    */
```
Comment (Dhors) [inline frontend/src/Components/Table.jsx:44]: small nits: 
 arbitary -> arbitrary 
"This variable changes is updated" -> This variable is updated

**Review by Dhors — APPROVED** [review]
looks good, nice work Kevin!

I added just a few notes added about small spelling errors and possible grammatical changes in comments, but nothing major.

File: frontend/src/Components/Table.jsx (hunk 2 above)
Comment (jge385) [reply frontend/src/Components/Table.jsx:44]: Thank u~

File: frontend/src/Hooks/useGameState.js [hunk 3]
```diff
@@ -9,13 +9,13 @@ const initialState = {
     dayPeriod: 'Day', // night or day
     dayNumber: 1, // what day it is
     alivePlayers: [],
-    status: '',
+    status: '', // someone is dead, or someone is on trail etc
```
Comment (reeved) [inline frontend/src/Hooks/useGameState.js:12]: trail -> trial

File: frontend/src/Hooks/useGameState.js [hunk 4]
```diff
@@ -9,13 +9,13 @@ const initialState = {
...
     winningRole: '',
     winners: [],
-    phase: '',
-    role: '',
-    amIDead: false,
-    checkedPlayers: [],
+    phase: '', // night-start, night-end, day-start, discussion-end etc
+    role: '', // detective or mafia or jester or civilian
+    amIDead: false, // the players themselves are dead or not
+    checkedPlayers: [], // the players whose role is known by other player
```
Comment (reeved) [inline frontend/src/Hooks/useGameState.js:18]: "list of players that the detective has checked"

File: frontend/src/Socket.js [hunk 5]
```diff
@@ -2,6 +2,8 @@ import socketIOClient from 'socket.io-client';
 
 /* eslint-disable new-cap */
 const socket = new socketIOClient('mafia-hands-backend-cd.herokuapp.com');
```
Comment (reeved) [inline frontend/src/Socket.js:4]: Could you add a comment saying: 

"Uncomment if you would like to use the deployed server"

File: frontend/src/Socket.js [hunk 6]
```diff
@@ -2,6 +2,8 @@ import socketIOClient from 'socket.io-client';
...
+
+// uncomment this for backend development
```
Comment (reeved) [inline frontend/src/Socket.js:6]: Change this to:
"Uncomment if you would like to use a local server to test server changes"

**Review by reeved — CHANGES_REQUESTED** [review]
Looking Good!
Just left some comments on some stuff for minor fixes :)

**Review by reeved — APPROVED** [review]
LGTM

**Pull request merged**